src/taskgraph/util/taskcluster.py

```
@@ -140,8 +140,26 @@ def get_session():
     return requests_retry_session(retries=5)
 
 
-def get_artifact_url(task_id, path):
-    artifact_tmpl = liburls.api(get_root_url(), "queue", "v1", "task/{}/artifacts/{}")
+def get_artifact_url(task_id, path, use_proxy=False):
+    if use_proxy:
```

I wonder if we should bypass this flag if `TASKCLUSTER_PROXY_URL` is not in os.environ

---

-1.  IMO if use_proxy is requested and we don't have a proxy, we should fail rather than silently not use it.

---

I think it would be better to raise the error as instead if they are expecting a proxy to be use but the environment variable isn't set I think we should throw an error